server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -1797,7 +1830,38 @@ export class AgenticChatController implements ChatHandlers {
         }
     }
 
+    #getToolOverWritableStream(
+        chatResultStream: AgenticChatResultStream,
+        toolUse: ToolUse
+    ): WritableStream | undefined {
+        const toolMsgId = toolUse.toolUseId!
+
+        return new WritableStream({
```

Comment:
How's this different from the stream used for executeBash? It seems to do almost the same thing, can we move the common logic to a util function?

Reply:
The stream for executeBash cannot be used by any other tool as everything is written as a shell command, it is very specific to executeBash use case.

I have added this simple generic stream that can be used by any tool to show updates.